docs/source/contrib_guide_code.rst

```diff
@@ -0,0 +1,103 @@
+==================================
+Contributing to the Code
+==================================
+A Note on Contributing to Open Source
+--------------------------------------
+Contributing to open source can be a nerve-wrecking process, but don't worry
+everyone on the Jupyter team is dedicated to making sure that your open source
+experience is as fun as possibly. At any time during the process described below,
+you can reach out to the Jupyter team on Gitter or the mailing list for
+assistance. If you are nervous about asking questions in public, you can also
+reach out to one of the Jupyter developers in private. You can use the public
+Gitter to find someone who has the best knowledge about the code you are working
+with and interact with the in a personal chat.
+
+As you begin your open source journey, remember that it's OK if you don't
+understand something, it's OK to make mistakes, and it's OK to only contribute
+a small amount of the code necessary to fix the issue you are tackling. Any and
+all help is welcome and any and all people are encouraged to contribute.
+
+How can I help?
+---------------
+Individuals are welcome, and encouraged, to submit pull requests and contribute
+to the Jupyter source. If you are a first-time contributor looking to get
```

Comment:
As a new contributer with a bit of github experience I think I would be confused by getting advise not to search the current repo but some complicated search string. I think it would be nice to add a small sentence along the lines of "the jupyter ecosystem is split across multiple repos. This can make it a bit complicated to find interesting issues to work on, so here is a way to find something".